test/autorest/generated/bytegroup/byte.go

```diff
@@ -1,158 +1,112 @@
-func DefaultByteClientOptions() ByteClientOptions {
-	return ByteClientOptions{
-		HTTPClient: azcore.DefaultHTTPClientTransport(),
-		Retry:      azcore.DefaultRetryOptions(),
-	}
-}
-
-// NewByteClient creates an instance of the ByteClient client.
-func NewByteClient(options *ByteClientOptions) (*ByteClient, error) {
-	return NewByteClientWithEndpoint("http://localhost:3000", options)
+// ByteOperations contains the methods for the Byte group.
+type ByteOperations interface {
+	// GetEmpty - Get empty byte value '' 
+	GetEmpty(ctx context.Context) (*ByteGetEmptyResponse, error)
+	// GetInvalid - Get invalid byte value ':::SWAGGER::::' 
+	GetInvalid(ctx context.Context) (*ByteGetInvalidResponse, error)
+	// GetNonASCII - Get non-ascii byte string hex(FF FE FD FC FB FA F9 F8 F7 F6) 
+	GetNonASCII(ctx context.Context) (*ByteGetNonASCIIResponse, error)
+	// GetNull - Get null byte value 
+	GetNull(ctx context.Context) (*ByteGetNullResponse, error)
+	// PutNonASCII - Put non-ascii byte string hex(FF FE FD FC FB FA F9 F8 F7 F6) 
+	PutNonASCII(ctx context.Context, byteBody []byte) (*BytePutNonASCIIResponse, error)
 }
 
-// NewByteClientWithEndpoint creates an instance of the ByteClient client.
-func NewByteClientWithEndpoint(endpoint string, options *ByteClientOptions) (*ByteClient, error) {
-	if options == nil {
-		o := DefaultByteClientOptions()
-		options = &o
-	}
-	p := azcore.NewPipeline(options.HTTPClient,
-		azcore.NewTelemetryPolicy(options.Telemetry),
-		azcore.NewUniqueRequestIDPolicy(),
-		azcore.NewRetryPolicy(&options.Retry),
-		azcore.NewRequestLogPolicy(options.LogOptions))
-	return NewByteClientWithPipeline(endpoint, p)
+type byteOperations struct {
+	*Client
+	azinternal.ByteOperations
 }
 
-// NewByteClientWithPipeline creates an instance of the ByteClient client.
-func NewByteClientWithPipeline(endpoint string, p azcore.Pipeline) (*ByteClient, error) {
-	u, err := url.Parse(endpoint)
-	if err != nil {
-		return nil, err
-	}
-	return &ByteClient{u: u, p: p}, nil
-}
-
-// GetEmpty get empty byte value ''
-func (client *ByteClient) GetEmpty(ctx context.Context) (*GetEmptyResponse, error) {
-	req, err := client.s.GetEmptyCreateRequest(*client.u)
+func (client *byteOperations) GetEmpty(ctx context.Context) (*ByteGetEmptyResponse, error) {
+	req, err := client.GetEmptyCreateRequest(*client.u)
 	if err != nil {
 		return nil, err
 	}
 	resp, err := client.p.Do(ctx, req)
 	if err != nil {
 		return nil, err
 	}
-	ba, err := client.s.GetEmptyHandleResponse(resp)
+	result, err := client.GetEmptyHandleResponse(resp)
 	if err != nil {
 		return nil, err
 	}
-	return ba, nil
+	return result, nil
 }
 
-// GetInvalid get invalid byte value ':::SWAGGER::::'
-func (client *ByteClient) GetInvalid(ctx context.Context) (*GetInvalidResponse, error) {
-	req, err := client.s.GetInvalidCreateRequest(*client.u)
+func (client *byteOperations) GetInvalid(ctx context.Context) (*ByteGetInvalidResponse, error) {
```

Comment:
I think the comments aren't being added.

Reply:
I was only emitting them on the method definitions in the interface since that's what will show up via intellisense.  I can add them here too if you think it's useful (I was on the fence about it).

Reply:
Hmm I think the linter will likely complain about them missing, I'll add them.

Reply:
Hmm, good point. My thought process was: if I'm a user and (have decided to ignore intellisense) I'm looking through the bytegroup code to see what is happening in GetInvalid and (for some reason) I dont understand what's happening there then I'd have to scroll back up to the method definition...which is fine...but possibly annoying? So I'm a little more for having the comments here as well.